crates/sui-core/src/authority/authority_store.rs

```diff
@@ -1630,14 +1636,19 @@ impl AuthorityStore {
         let cur_time = Instant::now();
         let mut pending_objects = vec![];
         let mut count = 0;
+        let mut size = 0;
         let package_cache = PackageObjectCache::new(self.clone());
         let (mut total_sui, mut total_storage_rebate) = thread::scope(|s| {
             let pending_tasks = FuturesUnordered::new();
             for o in self.iter_live_object_set(false) {
                 match o {
                     LiveObject::Normal(object) => {
-                        pending_objects.push(object);
+                        size += match &object.data {
+                            Data::Move(move_obj) => move_obj.contents().len(),
+                            Data::Package(move_pkg) => move_pkg.size(),
```

Comment:
is this the same size that is used for storage cost computation? also this doesn't count coins, does it?

Reply:
nevermind on the coins, i remembered that the coin storage format is converted to a move object when being loaded

Reply:
Yes, for package the storage cost is determined based on this function `object_size_for_gas_metering` which literally calls `.size()` for move_pkg. I think I could just call `object_size_for_gas_metering` on the object itself which makes it even simpler